citadel-internal-service/tests/service.rs

```diff
@@ -441,4 +443,158 @@ mod tests {
+                            cid,
+                            peer_cid,
+                            peer_username: _,
+                            request_id: _,
+                        },
+                    ) => {
+                        assert_eq!(cid, *cid_b);
+                        assert_eq!(peer_cid, *cid_a);
+                    }
+                    _ => {
+                        panic!("Peer B didn't get the PeerRegisterNotification, instead got {inbound_response:?}");
+                    }
+                }
+
+                // Service B Sends Register Request to Accept
+                to_service_b
+                    .send(InternalServiceRequest::PeerRegister {
+                        request_id: Uuid::new_v4(),
+                        cid: *cid_b,
+                        peer_cid: (*cid_a),
+                        session_security_settings,
+                        connect_after_register: false,
+                    })
+                    .unwrap();
+
+                // Receive Register Success Responses
+                let _ = from_service_a.recv().await.unwrap();
+                let _ = from_service_b.recv().await.unwrap();
+
+                // Service A Requests To Connect
+                to_service_a
+                    .send(InternalServiceRequest::PeerConnect {
+                        request_id: Uuid::new_v4(),
+                        cid: *cid_a,
+                        peer_cid: *cid_b,
+                        udp_mode: Default::default(),
+                        session_security_settings,
+                    })
+                    .unwrap();
+
+                // Service B Receives Connect Request from Service A
+                let inbound_response = from_service_b.recv().await.unwrap();
+                match inbound_response {
+                    InternalServiceResponse::PeerConnectNotification(PeerConnectNotification {
+                        cid,
+                        peer_cid,
+                        session_security_settings: _,
+                        udp_mode: _,
+                        request_id: _,
+                    }) => {
+                        assert_eq!(cid, *cid_b);
+                        assert_eq!(peer_cid, *cid_a);
+                    }
+                    _ => {
+                        panic!("Peer B didn't get the PeerConnectNotification");
+                    }
+                }
+
+                // Service B Sends Connect Request to Accept
+                to_service_b
+                    .send(InternalServiceRequest::PeerConnect {
+                        request_id: Uuid::new_v4(),
+                        cid: *cid_b,
+                        peer_cid: *cid_a,
+                        udp_mode: Default::default(),
+                        session_security_settings,
+                    })
+                    .unwrap();
+
+                // Receive Connect Success Responses
+                let _ = from_service_a.recv().await.unwrap();
```

Comment:
Here and everywhere else, let's make assertions that these signals are what they are supposed to be, that way, if we change anything in the future, we ensure a consistent flow.